lib/user_session_management.rb

```diff
@@ -121,6 +121,12 @@ def get_client_app
                       Doorkeeper::Application.find_by(id: session[:client_app])
   end
 
+  def is_redirect_url?(application:, url:)
+    return false if application.nil? || url.nil?
+    # Let doorkeeper do the work of checking the URL against the app's redirect_uris
+    Doorkeeper::OAuth::Helpers::URIChecker.valid_for_authorization?(url, application.redirect_uri)
+  end
```

Comment:
Guessing you moved this method here from `ApplicationController` so it could be available in the spec, but this method doesn't belong in this module.  This module is about accessing the session for user stuff.  Can you move this method to some other library file or even monkey patch the doorkeeper application model with `is_redirect_url?(url)`?